Cloud/main.py

```diff
@@ -1,7 +1,110 @@
 #!usr/bin/env python3
 #-*-coding:utf-8-*-
 
+import psycopg2
+import time
+import urllib3
+from edgeRequest import getTemperature
+
+dbname="Cloud"
+username_db="root"
+password_db="root"
+
+default_Location="Paris"
+measure_rate=60
 def run():
-    print("hello world!")
+    urllib3.disable_warnings()
+    connexion = init_db()
+    house_id = getHouseID(connexion)
+
+    print("Starting aquisition on edge, House n"+str(house_id))
+    while(True):
+
+        stats= getTemperature()
+        insert_data(connexion, house_id ,stats)
+        time.sleep(measure_rate)
+    print("Stopping aquisition")
+
+
+        
+    
+
+def init_db():
+    
+   
+    conn = psycopg2.connect(host="localhost",database=dbname, user=username_db, password=password_db)
+    db_cursor = conn.cursor()
+    
+    commands = (
+    """
+        CREATE TABLE IF NOT EXISTS Houses (
+            house_id SERIAL PRIMARY KEY,
+            location VARCHAR(255) NOT NULL
+        )
+        """,
+        """
+        CREATE TABLE IF NOT EXISTS GeoTemp (
+            measure_id SERIAL PRIMARY KEY,
+            house_id SERIAL,
+            temperature_avg FLOAT(5) NOT NULL,
+            temperature_unit CHAR,
+            delta FLOAT(5),
+            clim_duration FLOAT(2),
+            heat_duration FLOAT(2),
+            date VARCHAR(50),
+            FOREIGN KEY (house_id) REFERENCES Houses (house_id)
+        )
+        """)
+       
+    for command in commands:
+        db_cursor.execute(command)
+    db_cursor.close()
+    conn.commit()
+
+    return conn
+
+def insert_data(conn, house_id, request):
+    db_cursor=conn.cursor()
+    temp_avg = request.json().get("average")
+    temp_delta= request.json().get("std_deviation")
+    clim_duration= request.json().get("clim_duration")
+    heat_duration= request.json().get("heat_duration")
+    year= request.json().get("year")
+    temperature_unit= request.json().get("temperature_unit")
+    
+
+    command= """INSERT INTO GeoTemp(house_id, temperature_avg, temperature_unit, delta,clim_duration,heat_duration,date)
+             VALUES("""+str(house_id)+", "+str(temp_avg)+", '"+temperature_unit+"', "+str(temp_delta)+", "+str(clim_duration)+", "+str(heat_duration)+", "+year+") RETURNING measure_id;"""
+
+    db_cursor.execute(command)
+
+    measure_id= db_cursor.fetchone()[0]
+
+    conn.commit()
+    db_cursor.close()
+    
+
+    return measure_id
+
+def getHouseID(conn):
+    db_cursor=conn.cursor()
+    command ="""SELECT house_id FROM Houses"""
+    db_cursor.execute(command)
+    house_id=None
+    house_id_row = db_cursor.fetchone() #just taking the first one
+    if(house_id_row == None):
+        command =  """INSERT INTO Houses(location)
+             VALUES('"""+default_Location+"') RETURNING house_id;"

```

Comment:
Consider using standard SQL syntax with psycopg2 to format properly arguments like here : https://www.postgresqltutorial.com/postgresql-python/insert/
https://wiki.postgresql.org/wiki/Psycopg2_Tutorial
https://github.com/ttgc/TtgcBot/blob/v2.3.0/src/tools/BotTools.py

Reply:
I already seen the first website, but filling out the string from the start seem much more understandable than placing a placeholder that another function will use afterward